aspects/schema.go

```diff
@@ -669,6 +735,20 @@ func (v *stringSchema) Validate(raw []byte) (err error) {
 	return nil
 }
 
+// AtPath return the string schema if the path terminates at this schema and
+// an error if not.
+func (v *stringSchema) AtPath(path []string) ([]Schema, error) {
+	if len(path) != 0 {
+		return nil, fmt.Errorf(`cannot follow path beyond "string" type`)
```

Comment:
shouldn't these errors include path?

Reply:
They can't, the `path` variable is just a suffix at this point. That why previously the method's signature included the whole path and then an index (to go down the path parts). So this error has to be included in something else that carries the rest of the information. `aspects.go` check uses this to create something like `path "a.b.c" for request "req" is invalid: cannot follow path beyond "string" type` (line 436)

Reply:
maybe we should have a Error struct for this that capture the path suffix, so that the higher level can look at how long the suffix is, for example if the string is at "a.b",  the error would contain ["c"¨], so we could produce an error like:

path "a.b.c" for request "req" is invalid after "a.b":  cannot follow path beyond "string" type 

perhaps?  am I making sense?  the issue with the current error is that is not clear where the string is

Reply:
Done, added a custom error for this and expanded a test slightly to test it better